tools/autograd/derivatives.yaml

```diff
@@ -938,8 +941,8 @@
   self: zeros_like(grad)
 
 - name: sparse_mask(Tensor self, Tensor mask)
-  self: not_implemented("sparse_mask")
-  mask: not_implemented("sparse_mask")
+  self: grad.to_dense().sparse_mask(mask).to_dense()
```

Comment:
Why is this necessary to implement `to_sparse` backward?
Also, this (`sparse_mask` backward) needs a test given that you've implemented it.

Reply:
in `./aten/src/ATen/native/TensorConversions.cpp` you can see that to_dense_backward calls sparse_mask, which ends up causing this to be required.

Reply:
why do you need to densify the grad?  Modulo dealing with coalescing (does this do that correctly?), isn't the just creating a sparse tensor with the values of grad correct?

Reply:
sparse_mask requires a dense tensor and sparse mask. if I don't call to_dense() on the grad, iirc, sparse mask rejects it. If I don't call to_dense on the result, IIRC I got an error that the gradient layout has to match. I guess we could write an explicit kernel that took a sparse gradient and returned a dense mask though. I can add one, or at least file a ticket if you think it's worthwhile?